File: Data Structures & Algorithms/two-integer-sum/submission-5.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hm = {}

        for i in range(len(nums)):
            if nums[i] in hm.keys():
                return list([hm.get(nums[i]), i])

            hm[target - nums[i]] = i

        return null

File: Data Structures & Algorithms/search-2d-matrix/submission-5.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int c = matrix[0].length - 1;
        int check = 0;
        for (int i = 0; i < matrix.length; i++) {
            if (matrix[i][c] == target) return true;
            else if (target < matrix[i][c]) {
                check = i;
                break;
            }
        }
        System.out.println(check);
        int l = 0;
        int r = c;
        while (l <= r) {
            int mid = (l + r) / 2;
            if (matrix[check][mid] == target) return true;
            else if (matrix[check][mid] > target) r = mid - 1;
            else l = mid + 1;
        }
        return false;
    }
}
